overlord/snapstate/storehelpers.go

```diff
@@ -664,17 +674,40 @@ func storeUpdatePlan(
 			return updatePlan{}, fmt.Errorf("internal error: snap %q not found", sar.InstanceName())
 		}
 
-		// TODO:COMPS: handle components here
+		currentComps, err := snapst.CurrentComponentInfos()
+		if err != nil {
+			return updatePlan{}, err
+		}
+
+		compNames := make([]string, 0, len(currentComps))
+		for _, comp := range currentComps {
+			compNames = append(compNames, comp.Component.ComponentName)
+		}
+
+		// TODO:COMPS: handle components losing a resource that is currently
+		// installed
+		compTargets, err := componentTargetsFromActionResult(sar, compNames)
+		if err != nil {
+			return updatePlan{}, fmt.Errorf("cannot extract components from snap resources: %w", err)
+		}
+
+		// if we still have no channel here, this means that we refreshed
+		// by-revision without specifying a channel. make sure we continue to
```

Comment:
as we discussed is a bit unclear whether we should do this if revision was passed in explicitly from the user

Reply:
I think we still want to continue tracking that channel, right? We just don't want to send it to the store in the case that the user specified a channel. That is consistent with what we do right now.